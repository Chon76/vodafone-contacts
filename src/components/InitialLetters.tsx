import { HStack, Text } from 'native-base';
import { Pressable } from 'react-native';

type InitialsProps = {
  initials: Set<string>;
  selected: string;
  updateSelected: (letter: string) => void;
};

export default function InitialLetters({ initials, selected, updateSelected }: InitialsProps) {
  return (
    <HStack flexWrap={'wrap'} space={4} justifyContent={'center'}>
      {[...initials].sort().map((initial) => (
        <Pressable key={initial} onPress={() => updateSelected(initial)}>
          <Text fontWeight={'bold'} fontSize={'2xl'} color={initial === selected ? 'black' : 'gray.500'}>
            {initial}
          </Text>
        </Pressable>
      ))}

      <Pressable onPress={() => updateSelected('')}>
        <Text fontWeight={'bold'} fontSize={'2xl'} color={selected === '' ? 'black' : 'gray.500'}>
          [ALL]
        </Text>
      </Pressable>
    </HStack>
  );
}
