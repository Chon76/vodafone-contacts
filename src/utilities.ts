import { Contact } from './types/Contact';

// eslint-disable-next-line import/prefer-default-export
export function getInitialLeters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  console.log('initials', initialLetters);

  return initialLetters;
}
