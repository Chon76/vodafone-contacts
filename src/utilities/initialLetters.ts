import { Contact } from '../types/Contact';

/**
 *
 * @param contacts an array of {@link Contact}
 * @returns the initial letters of the names of the contact
 */
// eslint-disable-next-line import/prefer-default-export
export function getInitialLetters(contacts: Contact[]): Set<string> {
  const initialLetters: Set<string> = new Set();

  contacts.forEach((contact) => {
    initialLetters.add(contact.name.first[0].toUpperCase());
    initialLetters.add(contact.name.last[0].toUpperCase());
  });

  return initialLetters;
}
