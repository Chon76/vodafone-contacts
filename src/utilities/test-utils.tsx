/* eslint-disable import/no-extraneous-dependencies */
import { render, RenderOptions } from '@testing-library/react-native';
import React, { ReactElement } from 'react';
import { NativeBaseProvider } from 'native-base';
import { NavigationContainer } from '@react-navigation/native';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

const wrapper = ({ children }: { children: React.ReactNode }) => {
  return (
    <NativeBaseProvider initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  );
};

// wrap tests with necessery providers
const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper, ...options });

export * from '@react-navigation/native';
export { customRender as render };
