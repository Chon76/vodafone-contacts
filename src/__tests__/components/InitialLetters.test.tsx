import InitialLetters from '../../components/InitialLetters';
import { render } from '../../utilities/test-utils';

describe('initial letters', () => {
  it('renders correctly', () => {
    const inititals = new Set(['J', 'D', 'B', 'W']);

    const initialLetters = render(<InitialLetters initials={inititals} selected={''} updateSelected={() => {}} />);

    expect(initialLetters.toJSON()).toMatchSnapshot();
  });
});
