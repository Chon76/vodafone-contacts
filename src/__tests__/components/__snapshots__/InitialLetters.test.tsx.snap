// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initial letters renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={{}}
    style={
      {
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#71717a",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        B
      </Text>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "width": 16,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#71717a",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        D
      </Text>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "width": 16,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#71717a",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        J
      </Text>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "width": 16,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#71717a",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        W
      </Text>
    </View>
    <View
      dataSet={{}}
      style={
        {
          "width": 16,
        }
      }
    />
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
    >
      <Text
        dataSet={{}}
        style={
          {
            "backgroundColor": undefined,
            "color": "#000000",
            "fontFamily": undefined,
            "fontSize": 24,
            "fontStyle": "normal",
            "fontWeight": "700",
            "letterSpacing": 0,
            "lineHeight": 36,
            "textDecorationLine": undefined,
          }
        }
      >
        [ALL]
      </Text>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
